### PR TITLE
Homework #1: Tree

### PR DESCRIPTION
**Pull request by BThero**
Homework #1 - Tree

### CODE REVIEW

**Comment by korzio** [comment]
oops, too much removed? I can see zero changes

**Comment by BThero** [comment]
Finally it works

File: hw1/tree.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+if (args.length === 0) {
+	throw new Error('Please pass a .json file as an argument');
+}
+
+if (args.length > 1) {
+	throw new Error(`Too many arguments: expected 1, found ${args.length}`);
+}
+
+const data = JSON.parse(process.argv[2]);
```
Comment (korzio) [inline hw1/tree.js]: did you want to pass a path to a json file instead?
now to make it work I have to pass the whole json structure as a string argument, smth like

```bash
node tree '{"name":"abc","items":[{"name":"abcde","items":[{"name":"some long name","items":[{"name":"short","items":[{"name":"cool"},{"name":"good"}]}]},{"name":4}]},{"name":5,"items":[{"name":6}]}]}'
```
?

File: hw1/tree.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/* Usage: node tree [.json file] */
+
+const args = process.argv.slice(2);
+
+if (args.length === 0) {
+	throw new Error('Please pass a .json file as an argument');
+}
+
+if (args.length > 1) {
```
Comment (korzio) [inline hw1/tree.js:9]: nice, input validation!

File: hw1/tree.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+	if (!data.name) {
+		throw new Error('Error: no "name" property in the object');
+	}
+
+	let str = '';
+
+	for (let i = 0; i < depth; i++) {
+		if (i + 1 < depth) {
+			if (unclosed.some((x) => x == i)) {
```
Comment (korzio) [inline hw1/tree.js:33]: a little bit repeating your self on :37, maybe to have a separate function for it, also not clear from a reader pointer of view what's going on in code in this "if"

**Review by korzio — APPROVED** [review]

**Pull request merged**